client/js/components/ratings/rating.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import ActionGrade from 'material-ui/lib/svg-icons/action/grade';
+import RatingActions from '../../actions/rating-actions';
+import Star from './star';
+import UserStore from '../../stores/user-store';
+
+export default class Rating extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleMouseOver = this.handleMouseOver.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleMouseLeave = this.handleMouseLeave.bind(this);
+    this.state = {grade: 0, currentUser: UserStore.state.currentUser};
+    this.style = {
+      starBlack: {
+        fill: 'rgba(135, 135, 135, 0.8)'
+      },
+      starYellow: {
+        fill: 'rgba(255, 210, 30, 1)'
+      }
+    };
+  }
+
+  componentWillMount () {
+    this.setState({ currentUser: UserStore.state.currentUser });
+    RatingActions.getSnippetRatings(this.props.snippetId);
+    RatingActions.getCurrentUserRating(this.props.snippetId);
+  }
+
+  componentDidMount () {
+    this._setGradeOfUserRating();
+  }
+
+  handleMouseLeave(g){
+    this._setGradeOfUserRating();
+  }
+
+  handleMouseOver(g){
+    this.setState({grade: g});
+  }
+
+  handleClick(g){
+    RatingActions.create({
+      value: g,
+      SnippetId: this.props.snippetId
+    });
+    this.setState({grade: g});
+  }
+
+  _gradeToStyle(grade){
+    return grade > this.state.grade ? this.style.starBlack : this.style.starYellow;
+  }
+
+  _setGradeOfUserRating(){
+    var newGrade = 0;
+    if (this.props.usersRatings[this.state.currentUser.id]) {
```

Comment:
@dwoznicka It is only proposition to store value of ```this.props.usersRatings[this.state.currentUser.id]``` in ```const``` variable to improve code readability